slick/src/main/scala/slick/jdbc/RowsPerStatement.scala

```diff
@@ -0,0 +1,11 @@
+package slick.jdbc
+
+sealed trait RowsPerStatement
```

Comment:
I think what we should show in options is not how many lines to insert, but how they are implemented.

Currently implementation for Oracle is not supported in this PR, but it may need to provide more than one of several possible implementations.

https://database.guide/4-ways-to-insert-multiple-rows-in-oracle/

Reply:
> I think what we should show in options is not how many lines to insert, but how they are implemented.

I'm not clear what you're suggesting by this (or what it has to do with the rest of the comment)

Reply:
My feeling is that the number of rows is just not a very important concept.
The difference of these options is whether the implementation uses JDBC's batch API or the syntax of each database product. In the future, when Oracle is supported, there may be a difference between using INSERT ALL and Select Query.

Thus, I had the following names in mind as candidates

- Options
  - InsertOptions
  - StatementOptions
- Behaviors
- Mode
  - StatementMode

Reply:
> My feeling is that the number of rows is just not a very important concept.

I mean it is, it's the whole reason for this PR.

> The difference of these options is whether the implementation uses JDBC's batch API

That's not correct. Even with one-row-per-statement whether or not batching is used is a separate decision that's not controlled by this flag. It may not batch. The only thing that's certain in that case is that each row will involve a separate SQL statement.

> or the syntax of each database product. In the future, when Oracle is supported, there may be a difference between using INSERT ALL and Select Query.

I **think** you're alluding to the possibility of using this ADT to control which of the different ways of doing a single insert with all the rows in Oracle. I don't see the point of that. If we add support for single insert for Oracle we can just pick one method. Unless there are significant trade-offs between the different ways of achieving it, I don't think the user cares. But they do care that it's one insert statement with all the rows.

Plus, as discussed we may want to add an option for N rows per statement. So I really think it's about rows per statement, not anything else.

> 
> Thus, I had the following names in mind as candidates
> 
> * Options
>   
>   * InsertOptions
>   * StatementOptions
> * Behaviors
> * Mode
>   
>   * StatementMode

I feel like those are too vague. It's not obvious what they're for, which means users have to work harder to learn it, and library contributors have to work harder to not have name clashes and to know how to evolve things later.

Reply:
I understand your opinion, but my opinion remains the same.
However, it is only a matter of name, so either way is fine.

Reply:
> that's not controlled by this flag. It **may not** batch. 

Please understand that this is a story that could change the behavior of the existing ++=(= batch for each row).  All existing products using Slick are running in batch mode in all Profiles.